02-Generics/Generics/Generics.cs

```diff
@@ -91,9 +98,72 @@ public static class ArrayExtentions {
 		///     { 1, "a", false },
 		///   }
 		/// </example>
-		public static void SortTupleArray<T1, T2, T3>(this Tuple<T1, T2, T3>[] array, int sortedColumn, bool ascending) {
+		public static void SortTupleArray<T1, T2, T3>(this Tuple<T1, T2, T3>[] array, int sortedColumn, bool ascending) 
+				where T1: IComparable
+				where T2:IComparable
+				where T3:IComparable
+		{
```

Comment:
please rewrite it using Array.Sort(array, comparer)

note: to identify which column you'll need to use just create Func which return needed item :)

Reply:
good one, but lets try to rewrite without switch (and if-else) for getting needed value for sorting